cpp/include/cudf/detail/cuco_helpers.hpp

```diff
@@ -16,11 +16,17 @@
 
 #pragma once
 
+#include <cudf/types.hpp>
+
 #include <rmm/cuda_stream_view.hpp>
 #include <rmm/mr/device/polymorphic_allocator.hpp>
 
 namespace cudf::detail {
 
+/// Sentinel value for `cudf::size_type`
+cudf::size_type constexpr CUCO_CUDF_SIZE_TYPE_SENTINEL = -1;
+// TODO: is it a mouthful? Maybe `CUCO_SIZE_TYPE_SENTINEL`?
```

Comment:
Perhaps drop the CUCO?  `CUDF_SIZE_TYPE_SENTINEL`
I'm fine with any of these names.